docs/af5-app-from-scratch/modules/ROOT/pages/implement-command-subscribe-student.adoc

```diff
@@ -0,0 +1,601 @@
+----
+class SubscribeStudentCommandHandler {
+
+    // rest omitted for brevity
+
+    @EventSourcedEntity
+    static class State {
+
+        private CourseId courseId;
+        private int courseCapacity = 0;
+        private int noOfStudentsSubscribedToCourse = 0;
+
+        private StudentId studentId;
+        private int noOfCoursesStudentSubscribed = 0;
+        private boolean alreadySubscribed = false;
+
+        // other handlers added previously omitted for brevity
+
+        @EventSourcingHandler
+        void evolve(CourseCreated event) { // <1>
+            this.courseId = new CourseId(event.courseId());
+            this.courseCapacity = event.capacity();
+        }
+
+        @EventSourcingHandler
+        void evolve(CourseCapacityChanged event) { // <2>
+            this.courseCapacity = event.capacity();
+        }
+
+        @EventSourcingHandler
+        void evolve(StudentSubscribed event) { // <3>
+            var subscribingStudentId = new StudentId(event.studentId());
+            var subscribedCourseId = new CourseId(event.courseId());
+            if(subscribedCourseId.equals(courseId)) { // <4>
+                noOfStudentsSubscribedToCourse++;
+            }
+            if (subscribingStudentId.equals(studentId)) { // <5>
+                noOfCoursesStudentSubscribed++;
+            }
+            if (subscribingStudentId.equals(studentId) && subscribedCourseId.equals(courseId)) { // <6>
+                alreadySubscribed = true;
+            }
+        }
+
+        @EventSourcingHandler
+        void evolve(StudentUnsubscribed event) { // <7>
+            var subscribingStudentId = new StudentId(event.studentId());
+            var subscribedCourseId = new CourseId(event.courseId());
+            if(subscribedCourseId.equals(courseId)) {
+                noOfStudentsSubscribedToCourse--;
+            }
+            if (subscribingStudentId.equals(studentId)) {
+                noOfCoursesStudentSubscribed--;
+            }
+            if (subscribingStudentId.equals(studentId) && subscribedCourseId.equals(courseId)) {
+                alreadySubscribed = false;
+            }
+        }
+    }
+
+}
+----
+<1> Same as with a student, we store the `courseId` if was created along with the `capacity` from the same `CourseCreated` event.
+<2> We update the `capacity` on `CourseCapacityChanged` event.
+<3> In this case, we're going to evolve the State on every StudentSubscribed event related to the course or the student whose IDs are in the command.
+How we instruct the store to load those events, we will discuss in the next paragraph.
+For now, you need to be aware of that you may receive events about different student and different course as well, because we have one event handler per event type -> we handle there all `StudentSubscribed/Unsubscribed` for given course (different students) to know how much capacity left and all for given student (different courses) to know how many courses the student is subscribed to.
```

Comment:
This sentence doesn't flow clearly, but I am not sure how to adjust it without losing your intent. Could you please break this sentence up into several smaller sentences?